src/api/auth/auth.api.ts

```diff
@@ -18,23 +18,75 @@ export interface AuthSignUpDto {
   timeZone: string;
 }
 
+export interface AuthCheckEmailDto {
+  email: string;
+}
+
+export interface AuthForgotPasswordDto {
+  email: string;
+}
+
+export interface AuthResetPasswordDto {
+  token: string;
+  password: string;
+}
+
 export interface AuthLoginDto {
   email: string;
   password: string;
 }
 
+export interface AuthActivationDto {
+  link: string;
+}
+
 export const authAPI = {
   signUp(data: AuthSignUpDto) {
     return axios
       .post<AuthSignUpDto>(API_URLS.signUp, data)
-      .then((res) => res.data);
+      .then((res) => res.data)
+      .catch((error) => {
+        throw error.response.data;
+      });
+  },
+  checkEmail(data: AuthCheckEmailDto) {
```

Comment:
where is the return data type? have you checked the checklist?

Reply:
fixed